lib/test/clojure_lsp/classpath_test.clj

```diff
@@ -46,3 +48,116 @@
              {:project-path "bb.edn"
               :classpath-cmd ["bb" "print-deps" "--format" "classpath"]}]
             (classpath/default-project-specs #{:something :otherthing}))))))
+
+(defmacro never-skip-exec-tests
```

Comment:
I don't understand why we need all those macros and a new test suite, can't we just have integration tests for that? we already run integration tests on all platforms

Reply:
Sure, could you please point me to the relevant yaml file where integration test occur across all architectures so I can have a look and fit this in?

My logic for a separate file and the use of macros was this:
1. I considered these tests to be testing a single function in `lib`, thus I categorised them as unit tests.
2. I couldn't find anything relevant in `ci.yml` that would accommodate all existing unit tests + this one to run across all architectures; if I run the unit tests on MS-Windows (e.g. `cd lib && Clojure -M:test`), some tests fail (I plan to have a look at some point and fix them). Thus I decided to package this on a separate file so I can demonstrate the logic in a clear manner and receive feedback.
3. The tests are subject to the project executables to be installed on the local machine where the tests are run and being discoverable, they will otherwise fail.
3.1. I wanted users that run the test suite locally, not to fail these tests if the project exec (e.g. boot) was not locally installed, but skip it with a message. Thus, the user does not have to worry about an irrelevant test failing.
3.2. Likewise, I wanted the tests to properly fail in all circumstances when run in the CI environment and this can be done with the use of an env variable that is only set in the GH action.
3.3. And thus the use of macros to skip tests (there doesn't appear to be such a facility available just with `deftest) and test for an environment variable being set.

Thanks

Reply:
This is where we run the [integration-tests](https://github.com/clojure-lsp/clojure-lsp/blob/master/.github/workflows/ci.yml#L75), but I just noticed we don't run windows ones on every PR, maybe we should 🤔 I think it makes sense.

Reply:
I think we can keep it as unit tests too, but mock the existence of the commands on their machines so, otherwise those should be considered integration.
If so, we could remove the env var logic I think